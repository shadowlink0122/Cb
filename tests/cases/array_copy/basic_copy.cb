// 基本的な配列コピーテスト
int main() {
    int[3] arr1 = [1, 2, 3];
    int[3] arr2;
    
    // 手動コピー
    for (int i = 0; i < 3; i = i + 1) {
        arr2[i] = arr1[i];
    }
    
    // コピーされた値の検証
    if (arr2[0] != 1) {
        println("ERROR: arr2[0] should be 1, got %d", arr2[0]);
        return 1;
    }
    if (arr2[1] != 2) {
        println("ERROR: arr2[1] should be 2, got %d", arr2[1]);
        return 1;
    }
    if (arr2[2] != 3) {
        println("ERROR: arr2[2] should be 3, got %d", arr2[2]);
        return 1;
    }
    
    print(arr2[0]);
    print(arr2[1]);
    print(arr2[2]);
    
    println("Basic array copy test passed");
    return 0;
}
