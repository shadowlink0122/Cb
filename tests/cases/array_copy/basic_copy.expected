1
2
3
Basic array copy test passed
